app/controllers/facility_accounts_reconciliation_controller.rb

```diff
@@ -0,0 +1,96 @@
+class FacilityAccountsReconciliationController < ApplicationController
+
+  admin_tab :all
+  layout "two_column"
+
+  before_filter :authenticate_user!
+  before_filter :check_acting_as
+  before_filter :init_current_facility
+  before_filter :check_billing_access
+  before_filter :set_billing_navigation
+  before_filter { @accounts = account_class.need_reconciling(current_facility) }
+
+  def index
+    if @accounts.none?
+      # do nothing, just render
+    elsif selected_account
+      @unreconciled_details = unreconciled_details
+      @balance = selected_account.unreconciled_total(current_facility, @unreconciled_details)
+    else
+      redirect_to([account_route, :facility_accounts])
+    end
+  end
+
+  def update
+    update_account
+    redirect_to([account_route, :facility_accounts])
+  end
+
+  private
+
+  def set_billing_navigation
+    @subnav = "billing_nav"
+    @active_tab = "admin_billing"
+  end
+
+  def account_route
+    # This is coming in from the router, not the user, so it should be safe
+    Account.config.account_type_to_route(params[:account_type])
+  end
+  helper_method :account_route
+
+  def selected_account
+    @selected_account ||= if params[:selected_account].present?
+                            @accounts.find_by_id(params[:selected_account])
+                          else
+                            @accounts.first
+    end
+  end
+
+  def account_class
+    params[:account_type].constantize
+  end
+  helper_method :account_class
+
+  def unreconciled_details
+    OrderDetail
+      .account_unreconciled(current_facility, selected_account)
+      .order(%w(
+               order_details.account_id
+               order_details.statement_id
+               order_details.order_id
+               order_details.id))
+      .paginate(page: params[:page])
+  end
+
+  def update_account
```

Comment:
This method is pulled directly out of C2PO::FacilityAccountsControllerExtension. It could use some refactoring, but that seemed like more than I wanted to take on for this refactoring.